### ISSUE TITLE
Dev documentation: update title tag props comment or description to explain its usage

### ISSUE THREAD

**Issue by daine** [cds-snc/design-gc-conception#1300]
## 📇 User story
As a developer I want to know what the `notice-title-tag` and `card-title-tag` properties actually do so that I can use them effectively

## ✅ Definition of Done / Outcomes
- [ ] Done when it is clear that the `notice-title-tag` and `card-title-tag` properties only set the title tag to a different heading level for semantic / accessibility purposes, so that it appears in the correct order in the accessibility tree. It does not change the visual appearance of the title tag itself.

## 📜 Acceptance criteria
- [ ] Comments in the stencil prop have been updated
- [ ] README on the components affected have been updated (this should be auto generated)
- [ ] The documentation website guidance on the code mentions this clearly
- [ ] 

## 📝 More info
_Add here any context and or link that could be useful to understand the work_

**. 🚫 Out of scope**
_ If possible, give precisions on what is *not* part of this work_

**Comment by melaniebmn** [comment]
Component descriptions were done in [this PR](https://github.com/cds-snc/gcds-components/pull/745). Documentation site changes were done in [this PR](https://github.com/cds-snc/gcds-docs/pull/473).

Currently blocked by translation for the new documentation site content.
